### DISCUSSION TITLE
`{exp:http_header content_type="text/plain"}` should be excluded from the cache

### DISCUSSION THREAD

**Discussion by shbchk** [ExpressionEngine/ExpressionEngine#2875] [Feature Requests]
EE 7.2.5
PHP 7.4.21

How to reproduce:

1. Install the `HTTP Header` addon
2. Create a template with the following content:
```
{exp:http_header content_type="text/plain"}String 1
String 2
String 3
String 4
```
3. Enable template caching
4. Go to the rendered page. You will see correct output and correct "text/plain" header.

**Correct output for `text/plain` header:**
```
String 1
String 2
String 3
String 4
```

5. Refresh the page. You will see a different output and incorrect "text/html" header.

**Incorrect output for `text/html` header:**
```
String 1 String 2 String 3 String 4
```

6. Just to be sure - clear caches and repeat steps 4 and 5.

**Comment by intoeetive** [comment]
This happens because the plugin does not modify anything in the template, but is sending HTTP headers directly.

I have inspected possible options, and it does not seem like there's an easy solution here.

I suggest that you either use tag caching instead of template caching, or embed cached template into non-cached which would set the headers

**Comment by shbchk** [comment]
Good to know. Thank you, I will follow your recommendations!
